Subject: virtinst: Set is_onoff=True for iommufd
From: Nathan Chen nathanc@nvidia.com Fri Jun 12 21:18:38 2026 +0000
Date: Wed Sep 2 20:23:37 2026 +0200:
Git: 06f89b8d1dbee2d64737f0cffccae45b7568ff87

driver.iommufd=on emits invalid-looking iommufd="on" instead of
iommufd="yes". Add is_onoff=True to cli.py to address this.

Fixes: 9ab2918face14 ("virtinst: add support for iommufd")
Signed-off-by: Nathan Chen <nathanc@nvidia.com>

diff --git a/virtinst/cli.py b/virtinst/cli.py
index b5eef0ca7..db7279d02 100644
--- a/virtinst/cli.py
+++ b/virtinst/cli.py
@@ -5312,7 +5312,7 @@ class ParserHostdev(VirtCLIParser):
         cls.add_arg("type", "type")
         cls.add_arg("name", None, cb=cls.set_name_cb, lookup_cb=cls.name_lookup_cb)
         cls.add_arg("driver.name", "driver_name")
-        cls.add_arg("driver.iommufd", "driver_iommufd")
+        cls.add_arg("driver.iommufd", "driver_iommufd", is_onoff=True)
         cls.add_arg("rom.bar", "rom_bar", is_onoff=True)
         cls.add_arg("acpi.nodeset", "acpi_nodeset", can_comma=True)
         cls.add_arg("source.startupPolicy", "startup_policy")
